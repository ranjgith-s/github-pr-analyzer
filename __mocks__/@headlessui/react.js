// __mocks__/@headlessui/react.js
const React = require('react');

function passthrough({ children, ...props }) {
  return React.createElement('div', props, children);
}

module.exports = {
  __esModule: true,
  Card: passthrough,
  CardHeader: passthrough,
  CardBody: passthrough,
  CardFooter: passthrough,
  HeroUIProvider: passthrough,
};
