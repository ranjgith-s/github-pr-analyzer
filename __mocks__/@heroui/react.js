const React = require('react');

// List of custom props to filter out for DOM elements
const CUSTOM_PROPS = [
  'isCurrent',
  'endContent',
  'onPress',
  'variant',
  'color',
  'size',
  'underline',
  'src',
  'alt',
  'className',
  'children',
];

function filterProps(props) {
  const domProps = { ...props };
  CUSTOM_PROPS.forEach((key) => {
    if (key in domProps && key !== 'className' && key !== 'children') {
      delete domProps[key];
    }
  });
  return domProps;
}

function passthrough({ children, ...props }) {
  return React.createElement('div', { ...filterProps(props), className: props.className }, children);
}

function InputMock(props) {
  return React.createElement('input', filterProps(props));
}

function ButtonMock({ children, onPress, onClick, ...props }) {
  // Map onPress to onClick for DOM, do not pass onPress to DOM
  const handler = onPress || onClick;
  const domProps = { ...filterProps(props), onClick: handler, className: props.className };
  return React.createElement('button', domProps, children);
}

function Spinner() {
  // Only return a simple span for test, not a div with data-testid
  return React.createElement('span', null, 'Loading...');
}

function CardMock({ children, ...props }) {
  return React.createElement('div', { ...filterProps(props), className: (props.className || '') + ' mock-card' }, children);
}

module.exports = {
  __esModule: true,
  Card: CardMock,
  CardHeader: passthrough,
  CardBody: passthrough,
  CardFooter: passthrough,
  Button: ButtonMock,
  Input: InputMock,
  Avatar: passthrough,
  Breadcrumbs: passthrough,
  BreadcrumbItem: passthrough,
  Spinner,
};
