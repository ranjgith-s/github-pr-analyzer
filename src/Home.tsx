import React from 'react';
import { Box, Text } from '@primer/react';
import { Link as RouterLink } from 'react-router-dom';

export default function Home() {
  return (
    <Box display="flex" justifyContent="center" mt={6}>
      <Box
        as={RouterLink}
        to="/insights"
        sx={{
          p: 4,
          width: '100%',
          maxWidth: 400,
          borderWidth: '1px',
          borderStyle: 'solid',
          borderColor: 'border.default',
          borderRadius: 2,
          boxShadow: 'shadow.medium',
          textDecoration: 'none',
          color: 'inherit',
          '&:hover': { boxShadow: 'shadow.large', textDecoration: 'none' },
        }}
      >
        <Text fontSize={2} fontWeight="bold">
          Pull request insights
        </Text>
        <Text as="p" mt={2} color="fg.muted">
          Dive into metrics about your pull requests to track review timelines
          and lead time.
        </Text>
      </Box>
    </Box>
  );
}
