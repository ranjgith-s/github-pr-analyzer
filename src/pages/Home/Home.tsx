import React from 'react';
import { useDocumentTitle } from '../../hooks/useDocumentTitle';
import { useMetaDescription } from '../../hooks/useMetaDescription';
import {
  FolderIcon,
  UserGroupIcon,
  ArrowsRightLeftIcon,
} from '@heroicons/react/24/solid';
import { Card } from '@heroui/react';

export default function Home() {
  useDocumentTitle('PR-ism Home');
  useMetaDescription('Access GitHub pull request insights and metrics.');
  return (
    <div className="flex justify-center mt-12 gap-6">
      <Card
        as="a"
        href="/insights"
        className="p-8 w-full max-w-sm hover:shadow-xl transition-shadow"
      >
        <div className="flex items-center gap-2 mb-2">
          <ArrowsRightLeftIcon width={20} height={20} />
          <span className="text-lg font-bold">Pull Request Insights</span>
        </div>
        <p className="mt-4 text-foreground/60">
          See metrics for your pull requests, including review time and lead
          time.
        </p>
      </Card>
      <Card
        as="a"
        href="/developer"
        className="p-8 w-full max-w-sm hover:shadow-xl transition-shadow"
      >
        <div className="flex items-center gap-2 mb-2">
          <UserGroupIcon width={20} height={20} />
          <span className="text-lg font-bold">Developer Insights</span>
        </div>
        <p className="mt-4 text-foreground/60">
          View a developer&apos;s contributions and review activity across
          GitHub repositories with radar charts.
        </p>
      </Card>
      <Card
        as="a"
        href="/repo"
        className="p-8 w-full max-w-sm hover:shadow-xl transition-shadow"
      >
        <div className="flex items-center gap-2 mb-2">
          <FolderIcon width={20} height={20} />
          <span className="text-lg font-bold">Repository Insights</span>
        </div>
        <p className="mt-4 text-foreground/60">
          Explore repository health and DevOps metrics from GitHub.
        </p>
      </Card>
    </div>
  );
}
