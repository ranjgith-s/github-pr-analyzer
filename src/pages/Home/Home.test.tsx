import React from 'react';
import { render, screen } from '@testing-library/react';
import Home from './Home';

describe('Home', () => {
  it('renders Pull Request Insights card', () => {
    render(<Home />);
    const card = screen.getByText('Pull Request Insights').closest('[href]');
    expect(card).toBeInTheDocument();
    expect(card).toHaveAttribute('href', '/insights');
    expect(screen.getByText(/See metrics for your pull requests/i)).toBeInTheDocument();
  });

  it('renders Developer Insights card', () => {
    render(<Home />);
    const card = screen.getByText('Developer Insights').closest('[href]');
    expect(card).toBeInTheDocument();
    expect(card).toHaveAttribute('href', '/developer');
    expect(screen.getByText(/View a developer/i)).toBeInTheDocument();
  });

  it('renders Repository Insights card', () => {
    render(<Home />);
    const card = screen.getByText('Repository Insights').closest('[href]');
    expect(card).toBeInTheDocument();
    expect(card).toHaveAttribute('href', '/repo');
    expect(screen.getByText(/Explore repository health/i)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
