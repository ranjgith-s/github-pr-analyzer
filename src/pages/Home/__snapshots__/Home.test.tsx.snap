// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches snapshot 1`] = `
<div
  class="flex justify-center mt-12 gap-6"
>
  <a
    class="w-full max-w-sm"
    href="/insights"
  >
    <div
      class="rounded-xl border bg-card text-card-foreground shadow p-8 w-full hover:shadow-xl transition-shadow cursor-pointer"
    >
      <div
        class="flex items-center gap-2 mb-2"
      >
        <svg
          aria-hidden="true"
          data-slot="icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M15.97 2.47a.75.75 0 0 1 1.06 0l4.5 4.5a.75.75 0 0 1 0 1.06l-4.5 4.5a.75.75 0 1 1-1.06-1.06l3.22-3.22H7.5a.75.75 0 0 1 0-1.5h11.69l-3.22-3.22a.75.75 0 0 1 0-1.06Zm-7.94 9a.75.75 0 0 1 0 1.06l-3.22 3.22H16.5a.75.75 0 0 1 0 1.5H4.81l3.22 3.22a.75.75 0 1 1-1.06 1.06l-4.5-4.5a.75.75 0 0 1 0-1.06l4.5-4.5a.75.75 0 0 1 1.06 0Z"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="text-lg font-bold"
        >
          Pull Request Insights
        </span>
      </div>
      <p
        class="mt-4 text-foreground/60"
      >
        See metrics for your pull requests, including review time and lead time.
      </p>
    </div>
  </a>
  <a
    class="w-full max-w-sm"
    href="/developer"
  >
    <div
      class="rounded-xl border bg-card text-card-foreground shadow p-8 w-full hover:shadow-xl transition-shadow cursor-pointer"
    >
      <div
        class="flex items-center gap-2 mb-2"
      >
        <svg
          aria-hidden="true"
          data-slot="icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M8.25 6.75a3.75 3.75 0 1 1 7.5 0 3.75 3.75 0 0 1-7.5 0ZM15.75 9.75a3 3 0 1 1 6 0 3 3 0 0 1-6 0ZM2.25 9.75a3 3 0 1 1 6 0 3 3 0 0 1-6 0ZM6.31 15.117A6.745 6.745 0 0 1 12 12a6.745 6.745 0 0 1 6.709 7.498.75.75 0 0 1-.372.568A12.696 12.696 0 0 1 12 21.75c-2.305 0-4.47-.612-6.337-1.684a.75.75 0 0 1-.372-.568 6.787 6.787 0 0 1 1.019-4.38Z"
            fill-rule="evenodd"
          />
          <path
            d="M5.082 14.254a8.287 8.287 0 0 0-1.308 5.135 9.687 9.687 0 0 1-1.764-.44l-.115-.04a.563.563 0 0 1-.373-.487l-.01-.121a3.75 3.75 0 0 1 3.57-4.047ZM20.226 19.389a8.287 8.287 0 0 0-1.308-5.135 3.75 3.75 0 0 1 3.57 4.047l-.01.121a.563.563 0 0 1-.373.486l-.115.04c-.567.2-1.156.349-1.764.441Z"
          />
        </svg>
        <span
          class="text-lg font-bold"
        >
          Developer Insights
        </span>
      </div>
      <p
        class="mt-4 text-foreground/60"
      >
        View a developer's contributions and review activity across GitHub repositories with radar charts.
      </p>
    </div>
  </a>
  <a
    class="w-full max-w-sm"
    href="/repo"
  >
    <div
      class="rounded-xl border bg-card text-card-foreground shadow p-8 w-full hover:shadow-xl transition-shadow cursor-pointer"
    >
      <div
        class="flex items-center gap-2 mb-2"
      >
        <svg
          aria-hidden="true"
          data-slot="icon"
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 21a3 3 0 0 0 3-3v-4.5a3 3 0 0 0-3-3h-15a3 3 0 0 0-3 3V18a3 3 0 0 0 3 3h15ZM1.5 10.146V6a3 3 0 0 1 3-3h5.379a2.25 2.25 0 0 1 1.59.659l2.122 2.121c.14.141.331.22.53.22H19.5a3 3 0 0 1 3 3v1.146A4.483 4.483 0 0 0 19.5 9h-15a4.483 4.483 0 0 0-3 1.146Z"
          />
        </svg>
        <span
          class="text-lg font-bold"
        >
          Repository Insights
        </span>
      </div>
      <p
        class="mt-4 text-foreground/60"
      >
        Explore repository health and DevOps metrics from GitHub.
      </p>
    </div>
  </a>
</div>
`;
